### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by simahawk**
* cleanup and organize content
* add README generator info

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,25 +11,35 @@
...
 
 Goal: members of the teams should never be added directly on GitHub.
 They should be added on https://community.odoo.com. This script will
-sync all the teams from odoo to GitHub.
+sync all the teams from Odoo to GitHub.
 
 Prerequisites:
 
-* Your odoo user must have read accesses to the projects and users.
+* Your odoo user must have read accesses to the projects and users;
```
Comment (elicoidal) [inline README.md]:
```suggestion
* Your odoo user must have read access to the projects and users;
```

File: README.md [hunk 2]
```diff
@@ -100,6 +86,50 @@ OCA projects. It does not matter whether the forks exist on github or not, and
...
+
+These tools are meant to be used both by repo maintainers and contributors.
+You can leverage them to give more quality to your modules and to respect OCA guidelines.
+
+
+### README generator
+
+To provide high quality README for our modules we generate them automatically.
+The sections of the final README are organized in fragments.
+The must be put inside a `readme` folder respecting [this structure|./readme].
```
Comment (elicoidal) [inline README.md]:
```suggestion
They must be put inside a `readme` folder respecting [this structure|./readme].
```

File: README.md [hunk 3]
```diff
@@ -100,6 +86,50 @@ OCA projects. It does not matter whether the forks exist on github or not, and
...
+
+To generate the final README:
```
Comment (elicoidal) [inline README.md]:
```suggestion
eg.
To generate the final README for the module `auth_keycloak`:
```

File: README.md [hunk 4]
```diff
@@ -100,6 +86,50 @@ OCA projects. It does not matter whether the forks exist on github or not, and
...
+
+    $ oca-gen-addon-readme --repo-name=server-auth --branch=10.0 --addon-dir=auth_keycloak
+
+The result will be a fully PyPI compliant README.rst in the root of your module (`auth_keycloak` in this case).
```
Comment (elicoidal) [inline README.md]:
```suggestion
The result will be a fully PyPI compliant README.rst in the root of your module.
```

**Comment by simahawk** [comment]
@elicoidal integrated your changes on the fly (tried the new GH feature)

**Comment by sbidoul** [comment]
@simahawk can you squash? (I wish github would handle fixups automatically)

As a side not, IMO autopep8 should be removed from here now that black is a thing.

**Comment by simahawk** [comment]
@sbidoul done.
Regarding autopep8... I'm not sure... We should probably encourage ppl to use "black" but we should have at least some examples and proper configurations to share and then recommend it.

**Comment by sbidoul** [comment]
@simahawk if interested, you can look at https://github.com/acsone/click-odoo as an example of a modern code style enforcement, using black, isort and flake8. Start reading from the pre-commit config. Basically using black default config (it is not meant to be configured much anyway).

**Pull request merged**